parcels/field.py

```diff
@@ -935,12 +935,20 @@ def write(self, filename, varname=None):
         vname_depth = 'depth%s' % self.name.lower()
 
         # Create DataArray objects for file I/O
-        t, d, x, y = (self.grid.time.size, self.grid.depth.size,
-                      self.grid.lon.size, self.grid.lat.size)
-        nav_lon = xr.DataArray(self.grid.lon + np.zeros((y, x), dtype=np.float32),
-                               coords=[('y', self.grid.lat), ('x', self.grid.lon)])
-        nav_lat = xr.DataArray(self.grid.lat.reshape(y, 1) + np.zeros(x, dtype=np.float32),
-                               coords=[('y', self.grid.lat), ('x', self.grid.lon)])
+        t, d = (self.grid.time.size, self.grid.depth.size)
+        if type(self.grid) is RectilinearZGrid:
+            x, y = (self.grid.lon.size, self.grid.lat.size)
+            nav_lon = xr.DataArray(self.grid.lon + np.zeros((y, x), dtype=np.float32),
+                                   coords=[('y', self.grid.lat), ('x', self.grid.lon)])
+            nav_lat = xr.DataArray(self.grid.lat.reshape(y, 1) + np.zeros(x, dtype=np.float32),
+                                   coords=[('y', self.grid.lat), ('x', self.grid.lon)])
+        elif type(self.grid) is CurvilinearZGrid:
+            y, x = self.grid.lat.shape
```

Comment:
self.grid.xdim and self.grid.ydim exist (in a unique way, indpendent of Rectilinear and Curvi grids.
(also zdim)